Modules/_sqlite/cursor.c

```diff
@@ -436,7 +436,10 @@ begin_transaction(pysqlite_Connection *self)
     Py_BEGIN_ALLOW_THREADS
     sqlite3_stmt *statement;
     char begin_stmt[16] = "BEGIN ";
-    (void)strncat(begin_stmt, self->isolation_level, 9);
+    size_t len = strlen(self->isolation_level);
+    // bpo-45512: The length of char string should be well cared.
+    assert(len <= 9);
```

Comment:
`strlen` is only needed for `assert`, so we can wrap it in `#if Py_DEBUG`:

```suggestion
#ifdef Py_DEBUG
    size_t len = strlen(self->isolation_level);
    assert(len <= 9);
#endif
```